lib/TelemetrySender.js

```diff
@@ -1,19 +1,48 @@
 /* globals Services */
 
-const {Cu} = require("chrome");
+const {Cu, Cc, Ci} = require("chrome");
+
 Cu.import("resource://gre/modules/Services.jsm");
+Cu.importGlobalProperties(["XMLHttpRequest"]);
+
+function TelemetrySender(pingPref, pingEndpoint) {
+  this._pingPref = pingPref;
 
-function TelemetrySender() {
   Services.obs.addObserver(this, "tab-session-complete");
+  this._prefs = Cc["@mozilla.org/preferences-service;1"]
```

Comment:
an example in action: https://dxr.mozilla.org/mozilla-central/source/browser/components/nsBrowserGlue.js#408